File: LeetCode_Algorithm/374(Guess Number Higher or Lower)/374.py
class Solution:
    def guessNumber(self, n: int) -> int:
        # sol1 24ms 99% 13.9MB 16%
        left, right = 1, n
        while left <= right:
            mid = left + ((right-left) // 2)
            res = guess(mid)
            if res == 0:
                return mid
            elif res == -1:
                right = mid-1
            else:
                left = mid+1

File: LeetCode_Algorithm/374(Guess Number Higher or Lower)/374.js
var guessNumber = function(n) {
  // 87ms 65% 42.3MB 10.6%
  let left = 1; let right = n
  while (true) {
    let mid = left + parseInt((right-left) / 2)
    let res = guess(mid)
    if (res == -1) {
      right = mid-1 
    } else if (res == 1) {
      left = mid+1
    } else {
      return mid
    }
  }  
};